### PR TITLE
Feature: Validate conflicting autoLogin and verifyEmail flags

### LINKED ISSUE #1: Throw invalid config if `web.register.autoLogin` is true, while email verification is enabled
The property `stormpath.web.register.autoLogin` is `false` by default, but if the developer sets to `true` it means that we automatically create the Oauth2 cookies for the user after they register, an then redirect to the `nextUri`.

This is only possible if the email verification feature is disabled on the default account store of the specified application.

If the default account store has email verification enabled, and the developer defines `autoLogin` as `true`, we should throw an invalid configuration error:

> Invalid configuration: stormpath.web.register.autoLogin is true, but the default account store of the specified application has the email verification workflow enabled.  Auto login is only possible if email verification is disabled.  Please disable this workflow on this application's default account store.

### PR DESCRIPTION
**Pull request by typerandom**
Validates that auto login and email verification features aren't enabled at the same time.

Closes #1

### CODE REVIEW

**Comment by rdegges** [comment]
+100 awesome fix

File: lib/strategy/EnrichIntegrationFromRemoteConfigStrategy.js [hunk 1]
```diff
@@ -176,7 +177,7 @@ EnrichIntegrationFromRemoteConfigStrategy.prototype._enrichWithDirectoryPolicies
     extend(config, {
       web: {
         forgotPassword: {
-          enabled: config.web.changePassword.enabled = isEnabled(directory.passwordPolicy.resetEmailStatus)
```
Comment (timothyej) [inline lib/strategy/EnrichIntegrationFromRemoteConfigStrategy.js]: Is this change intentional? That `config.web.changePassword.enabled` is not set to whatever `config.web.forgotPassword.enabled` is set to anymore?

Comment (timothyej) [inline lib/strategy/EnrichIntegrationFromRemoteConfigStrategy.js]: @typerandom?

Comment (typerandom) [inline lib/strategy/EnrichIntegrationFromRemoteConfigStrategy.js]: No it is not. Thanks for noticing! Will fix that :)

Comment (timothyej) [inline lib/strategy/EnrichIntegrationFromRemoteConfigStrategy.js]: I fixed it for you, and I added some tests for it too :) Please review!

**Comment by typerandom** [comment]
@timothyej Looks good. Let's merge this! :)
